google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java

```diff
@@ -360,20 +393,87 @@ private void registerOpenTelemetryMetrics() {
             .setInstrumentationVersion(
                 ConnectionWorker.class.getPackage().getImplementationVersion())
             .build();
-    instrumentIncomingRequestCount =
+    telemetryMetrics.instrumentSentRequestCount =
+        writeMeter
+            .counterBuilder("append_requests_sent")
+            .setDescription("Counts number of requests sent over the network")
+            .build();
+    telemetryMetrics.instrumentSentRequestSize =
+        writeMeter
+            .counterBuilder("append_request_bytes_sent")
+            .setDescription("Counts byte size of requests sent over the network")
+            .build();
+    telemetryMetrics.instrumentSentRequestRows =
+        writeMeter
+            .counterBuilder("append_rows_sent")
+            .setDescription("Counts number of request rows sent over the network")
+            .build();
+    telemetryMetrics.instrumentAckedRequestCount =
         writeMeter
-            .counterBuilder("append_requests")
-            .setDescription("Counts number of incoming requests")
+            .counterBuilder("append_requests_acked")
```

Comment:
This can be a TODO, I am wondering if it is possible to have a Retry attribute to the metric.

Reply:
Done